fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -3902,4 +3952,56 @@ public void closeAsMarkedForReschedule(final Loan loan, final JsonCommand comman
 
         loanTransactionValidator.validateLoanRescheduleDate(loan);
     }
+
+    private void calculatePartialPeriodInterest(final Loan loan, final LocalDate chargeOffDate) {
+        loan.getRepaymentScheduleInstallments().stream()
+                .filter(installment -> !installment.getFromDate().isAfter(chargeOffDate) && installment.getDueDate().isAfter(chargeOffDate))
+                .forEach(installment -> {
+                    final BigDecimal totalInterest = installment.getInterestOutstanding(loan.getCurrency()).getAmount();
+                    final long totalDaysInPeriod = ChronoUnit.DAYS.between(installment.getFromDate(), installment.getDueDate());
+                    final long daysTillChargeOff = ChronoUnit.DAYS.between(installment.getFromDate(), chargeOffDate);
+
+                    final BigDecimal interestTillChargeOff = totalInterest
+                            .divide(BigDecimal.valueOf(totalDaysInPeriod), 10, RoundingMode.HALF_UP)
```

Comment:
You might wanna use the Math context from Moneyhelper to be unified! Hardcoding values are not recommended!